pkg/diagnose/diagnose.go

```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package diagnose
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
+	"github.com/cihub/seelog"
+	"github.com/fatih/color"
+)
+
+// Diagnose runs some connectivity checks, output it in writer
+// and returns the number of failed diagnosis
+func Diagnose(w io.Writer) (int, error) {
```

Comment:
To avoid confusion with `Diagnose` from the `Diagnosis` interface, can you rename it `RunDiagnostics`, or just `RunAll` as it'll be called as `diagnose.RunAll()`?

Reply:
I'll go with `RunAll()` 👍